basket/news/tasks.py

```diff
@@ -717,6 +717,250 @@ def record_source_url(email, source_url, newsletter_id):
+    # if no user was found, create new user in sfdc
+    else:
+        contact_data['first_name'] = first
+        contact_data['last_name'] = last
+        contact_data['email'] = data['email']
+
+        # create the user in sfdc
+        sfdc.add(contact_data)
+
+
+@et_task
+def process_subhub_event_customer_updated(data):
+    """
+    Updates customer record in SFDC if payment processor/Stripe customer
+    changes their name or email address
+    """
+    statsd.incr('news.tasks.process_subhub_event.customer_updated')
+
+    first, last = split_name(data['name'])
+    contact_data = {}
+
+    # find user by their payment processor ID
+    user_data = get_user_data(payee_id=data['customer_id'])
+
+    # if user was found in sfdc, see if we should update their name(s)
+    if user_data:
+        # if current last name is '_', update it
+        if user_data['last_name'] == '_':
+            contact_data['last_name'] = last
+
+        # if current last name is blank/Null, update it
+        if not user_data['first_name']:
+            contact_data['first_name'] = first
+
+        if 'first_name' in contact_data or 'last_name' in contact_data:
+            sfdc.update(user_data, contact_data)
+    # if user wasn't found, add them
+    else:
+        contact_data['first_name'] = first
+        contact_data['last_name'] = last
+        contact_data['email'] = data['email']
+        contact_data['payee_id'] = data['customer_id']
+
+        # create the user in sfdc
+        sfdc.add(contact_data)
+
+
+@et_task
+def process_subhub_event_subscription_charge(data):
+    """
+    This method handles both new and recurring charges.
+
+    There are two events that trigger this task:
+    - invoice.finalized
+    - invoice.payment_succeeded
+
+    Each of these events contains different information on the charge which
+    will need to be combined.
+
+    The charge_id field (sent in data) will tie the two events together.
+
+    invoice.finalized *should* be sent first, which is missing the following
+    credit card related fields:
+
+    - brand
+    - last4
+    - exp_month
+    - exp_year
+
+    We also need to determine if this payment (the combination of both events
+    listed above) is the first or is a recurrence. This is done by checking to
+    see if the subscription_id in data already exists in Salesforce.
+    """
+
+    statsd.incr('news.tasks.process_subhub_event.subscription_charge')
+
+    user_data = get_user_data(payee_id=data['customer_id'])
+
+    # the only user identifiable information available is the payment
+    # processor/Stripe ID, so if the user wasn't found by that, there's really
+    # nothing to be done here
+    if user_data:
+        # this is the primary key for the opportunity
+        charge_id = data['charge_id']
+
+        # just in case amount data is missing/malformed
+        try:
+            # amount values are cents - convert to dollars
+            amount_paid = int(data['amount']) / 100
```

Comment:
This always returns `int`. 
```python
int(108.5) / 100 = 1
```
Just a heads up, not sure if that's what you want here.

Reply:
Hmmm, good catch. Will replace with:

`int(data['amount']) / float(100)`

That should do it, right?

Reply:
Yes that works. Another trick is

`int(data['amount']) / 100.0`